.husky/pre-commit

```diff
@@ -1,5 +1,12 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
+if [ -z "$(git diff --cached --ignore-blank-lines --unified=0 | grep -E '^[+-]\s' | grep -vE '^[+-]\s*$')" ]; then
+  echo "Error: commit must contain new content."
+  exit 1
```

Comment:
The commits with empty lines are generated by automated workflow scripts:

https://github.com/heyxyz/hey/blob/main/.github/workflows/i18n-extract.yml
https://github.com/heyxyz/hey/blob/main/.github/workflows/crowdin-download.yml

So if we just do a pre-commit hook, the scripts would fail, I assume?
I think that this issue needs to be solved by configuring the workflow scripts.

Reply:
it seems like  [crowdin-download.yml](https://github.com/heyxyz/hey/blob/main/.github/workflows/crowdin-download.yml) adds newlines to the end of .po files, and [i18n-extract.yml](https://github.com/heyxyz/hey/blob/main/.github/workflows/i18n-extract.yml) removes them again :)

Normally, a textfile should have a newline at the end, so I would suggest fixing it in i18n-extract.yml, but any way is OK IMO.

https://github.com/heyxyz/hey/issues/3807#issuecomment-1743650348

Reply:
Thanks for these insights, will do the needful asap :+1: